rust-mode.el

```diff
@@ -1746,6 +1748,37 @@ visit the new file."
       (let ((output (json-read)))
         (cdr (assoc-string "root" output))))))
 
+(defun rust-insert-dbg ()
+  "Insert the dbg! macro."
+  (insert-parentheses (unless (region-active-p) 1))
```

Comment:
this may fail on unbalanced expressions while inserting '(', like on `|(abcd => |((abcd`. To improve on that, you may need to use simpler library functions like forward-sexp instead.